src/main/java/io/github/jadefalke2/CompoundCatchClause.java

```diff
@@ -0,0 +1,36 @@
+package io.github.jadefalke2;
+
+import io.github.jadefalke2.interfaces.CatchClause;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * A catch clause that maps many exception types to a single consumer.
+ */
+public class CompoundCatchClause implements CatchClause {
+    private final Set<Class<? extends Exception>> exceptionTypes;
+    private final Consumer<? super Exception> consumer;
+
+    public CompoundCatchClause(Collection<Class<? extends Exception>> exceptionTypes, Consumer<? super Exception> consumer) {
```

Comment:
For the sake of keeping this aligned with the actual Java catch blocks this should throw an exception if any one exception type is a subclass of another within this clause.

Reply:
this will add another `O(n²)` check though which might be too big of a trade off for such a check